tests/utils/test_urlutils.py

```diff
@@ -227,3 +230,32 @@ def test_url_host(self):
         """Test with an URL with no path."""
         url = QUrl('http://qutebrowser.org/')
         assert urlutils.filename_from_url(url) == 'qutebrowser.org.html'
+
+
+class TestSchemeOrder:
+
+    """Tests for scheme-order setting."""
+
+    def test_http(self, monkeypatch, config_stub):
+        """Test with scheme_order = http."""
+        init_config_stub(config_stub, 'naive', scheme_order=['http'])
+        monkeypatch.setattr('qutebrowser.utils.urlutils.config', config_stub)
+        assert (urlutils.fuzzy_url('example.com').toDisplayString() ==
+                'http://example.com')
+
+    def test_https(self, config_stub, monkeypatch):
+        """Test with scheme_order = https."""
+        init_config_stub(config_stub, 'naive', scheme_order=['https'])
+        monkeypatch.setattr('qutebrowser.utils.urlutils.config', config_stub)
+        assert (urlutils.fuzzy_url('example.com').toDisplayString() ==
+                'https://example.com')
+
+    def test_https_http(self, config_stub, monkeypatch):
+        """Test with scheme_order = https, http."""
+        init_config_stub(config_stub, 'naive', scheme_order=['https', 'http'])
+        monkeypatch.setattr('qutebrowser.utils.urlutils.config', config_stub)
+        assert (urlutils.fuzzy_url('example.com').toDisplayString() ==
+                'https://example.com')
+
+        # TODO: I don't know how to test the code in browser/webpage.py
+        # _handle_errorpage() ...
```

Comment:
Currently not at all - I'm working on tests for everything, so in a few weeks this should be much easier. For now, just test it by hand, and I'll write tests for it when I get to the `WebPage` tests. I'll notice it's untested because of test coverage anyways.